### ISSUE TITLE
Block editing theme-controlled template overrides unless LOCAL_THEME is set

### ISSUE THREAD

**Issue by benjaminjkraft** [learning-unlimited/ESP-Website#2545]
We have a magic tag that prevents clobbering template overrides for sites with custom themes.  But if you don't set it, well, then it doesn't work.  We should prevent doing anything likely to get clobbered unless you've set the setting.  This will also mean chapters have to talk to us before editing theme-controlled templates, which is probably a good idea to minimize the number of chapters whose custom themes we get stuck maintaining.

**Comment by benjaminjkraft** [comment]
Or, a more radical idea: we could whitelist template overrides.  Say you can only override printables, unless LOCAL_THEME is set.  Then we could have a lot more control over what is likely to break on a theme update.

**Comment by milescalabresi** [comment]
What are the theme-controlled templates? Would it be possible to automatically check the version of those TOs and toggle the tag iff the version number is greater than 1?

Something that gives me pause is that setting LOCAL_THEME = True disables access to the theme editor, which seems right for programs with custom themes but seems excessive for minor cosmetic changes to main.html (such as Cornell's change that they put button icons over the "My Profile" hyperlinks). I think it would be nice if they could make those minor changes and still be able to change their theme colors.

**Comment by willgearty** [comment]
@milescalabresi reminded me of #692

**Comment by benjaminjkraft** [comment]
Sorry, to be clear, I mean the template overrides that the theme edits.  We would just prohibit editing them, I think that's easier than trying to toggle the setting, since it's in settings, not tags.

For better or worse, it's really not great to have chapters making minor changes to themes, at least unless/until we come up with a better mechanism (e.g. #692).  There are a number of buttons in the theme editor (maybe literally all of them?) that can clobber those changes by accident, which is why LOCAL_THEME disables them.  So once you've edited your theme, it is not safe to change colors.  And, plus, it makes it more work for us to make stable releases -- template override diff view and LOCAL_THEME will help, but it's still a bunch of merging work.  If someone wants to make a minor change, they should just tell us to make it in their branch, or add a theme option, or things like that.

Note that #692 or #2046 would subsume this, as with those we could get rid of LOCAL_THEME most likely.

(And, to be clear, I think all template overrides are a bad idea.  I would love to get rid of them.  But we'd need to do #2046, and make some theme-like mechanism for having multiple printables template options, before we could do that, so that's more of just a vague notion of the direction I think is right, rather than a statement of plans.)

**Comment by milescalabresi** [comment]
Right, sorry, I was asking for the list. Is it just ``main.html``, ``index.html``, ``qsdfront.html``, and ``bubblesfront.html`` or are there more?

I don't disagree for most theme-releated things, but in defense of template overrides, I think they're useful for things like teacher/student schedules and other files that programs may want to customize.

**Comment by benjaminjkraft** [comment]
Ah I see!  There are a few more, depending on the theme -- it's everything in `esp/esp/themes/theme_data/*/templates`.

**Comment by abhinavjha0239** [comment]
Hi @benjaminjkraft @willgearty, I went through the issue and the related ones — would love to take this up. Can you assign it to me?

**Comment by willgearty** [comment]
I think this is actually resolved by #3618
